src/main/java/fruitshop/service/reader/FileReader.java

```
@@ -0,0 +1,8 @@
+package fruitshop.service.reader;
+
+import java.io.IOException;
+import java.util.List;
+
+interface FileReader {
+    List<String> read(String fileName) throws IOException;
```

This violates checklist item: 'Don't ignore exceptions in try-catch blocks'. The method signature throws IOException, but the requirements specify that IOExceptions should be caught and rethrown as RuntimeExceptions with informative messages and the original cause.